resteasy-spring-boot-test/src/main/java/com/paypal/springboot/resteasy/test/RestEasyApplicationTest.java

```diff
@@ -0,0 +1,75 @@
+package com.paypal.springboot.resteasy.test;
+
+import org.junit.AfterClass;
+import org.junit.Before;
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.context.ConfigurableApplicationContext;
+import org.springframework.util.SocketUtils;
+
+import io.restassured.RestAssured;
+
+/**
+ * Integration test for a resteasy service. This test will start the spring boot
+ * application in an embedded container and executes real http requests against
+ * it. The responses are verified in the test.<br>
+ *
+ * @author Christian Fehmer
+ *
+ */
+public class RestEasyApplicationTest {
+
+    private static ConfigurableApplicationContext context;
+    private int port;
+    private Class<?> applicationClass;
+    private String basePath;
+
+    /**
+     * initializes a test for the given application
+     *
+     * @param applicationClass
+     *            the main application class, most likely annotated with
+     *            {@link SpringBootApplication} and containing a static main
+     *            method with a <code>SpringApplication.run(...)</code>
+     *            statement.
+     * @param basePath
+     *            base path of the service under test, e.g. /sample-app/echo.
```

Comment:
I believe the example should be only "/sample-app". This sample application happens to have only one resource class, but it could have more, and with different Path annotations.